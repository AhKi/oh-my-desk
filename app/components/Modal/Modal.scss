.Modal__overlay {
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  bottom: 0;
  background-color: rgba(0,0,0,.5);
  z-index: 3;
}

.Modal__container {
  background-color: #F1F1F1;
  opacity: 1;
  position: absolute;
  margin: auto;
  top: 10%;
  left: 25%;
  width: 50%;
  height: 50%;
  z-index: 5;
}