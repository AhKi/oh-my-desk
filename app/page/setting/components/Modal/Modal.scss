.Modal__overlay {
  align-items: center;
  animation-duration: 0.2s;
  animation-name: RootModal__overlay;
  background-color: rgba(0,0,0,.6);
  display: flex;
  height: 100%;
  justify-content: center;
  left: 0;
  overflow: auto;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 3;
}

.Modal__container {
  animation-duration: 0.3s;
  animation-name: RootModal;
  background-color: $color-white;
  border-radius: 4px;
  box-sizing: inherit;
  display: flex;
  margin-bottom: 12px;
  overflow: auto;
  box-shadow: 0 60px 40px -30px rgba(0,0,0,.35);
  outline: none;
}

@keyframes RootModal {
  0% {
    opacity: 0;
    position: relative;
    top: -20px;
  }

  100% {
    opacity: 1;
    position: relative;
    top: 0;
  }
}

@keyframes RootModal__overlay {
  0% {
    opacity: 0;
  }

  100% {
    opacity: 1;
  }
}
