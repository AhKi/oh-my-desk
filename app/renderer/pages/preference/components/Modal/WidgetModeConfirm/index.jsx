import React from 'react';
import PropTypes from 'prop-types';
import i18n from 'constants/i18n';
import './WidgetModeConfirm.scss';

const propTypes = {
  onChangeMode: PropTypes.func,
  onModalClose: PropTypes.func,
};
const defaultProps = {
  onChangeMode() {},
  onModalClose() {},
};

class WidgetModeConfirm extends React.Component {
  constructor(props) {
    super(props);
    this.handleChange = this.handleChange.bind(this);
  }

  handleChange() {
    const { onChangeMode, onModalClose } = this.props;

    onChangeMode();
    onModalClose();
  }

  render() {
    const { onModalClose } = this.props;
    const text = i18n().preference;

    return (
      <div className="WidgetModeConfirm__container">
        <h3>{text.widgetModeConfirmTitle}</h3>
        <p className="WidgetModeConfirm__content">
          {text.widgetModeConfirmContent}
        </p>
        <div className="WidgetModeConfirm__btn-set">
          <button
            className="Btn Btn--sm Btn--gray"
            type="button"
            onClick={onModalClose}
          >
            {text.changedCancel}
          </button>
          <button
            className="Btn Btn--sm Btn--primary"
            type="button"
            onClick={this.handleChange}
          >
            {text.ok}
          </button>
        </div>
      </div>
    );
  }
}

WidgetModeConfirm.propTypes = propTypes;
WidgetModeConfirm.defaultProps = defaultProps;

export default WidgetModeConfirm;
