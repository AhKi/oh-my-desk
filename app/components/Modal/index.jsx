import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import MakeWebWidget from './Content/MakeWebWidget';
import './Modal.scss';

const propTypes = {
	modalType: PropTypes.string,
	modalProps: PropTypes.object, // eslint-disable-line
};

const defaultProps = {
	modalType: '',
	modalProps: {},
};

const content = {
	MAKE_WEB_WIDGET: MakeWebWidget,
};

class Modal extends React.Component {
	render() {
		const { modalType, modalProps } = this.props;
		const isOpen = !!modalType;
		const Content = content[modalType];

		if (!isOpen) {
			return null;
		}

		return (
			<ReactModal
				className="Modal__container"
				overlayClassName="Modal__overlay"
				contentLabel="Modal"
				isOpen={isOpen}
				ariaHideApp={false}
			>
				{<Content {...modalProps} />}
			</ReactModal>
		);
	}
}

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;