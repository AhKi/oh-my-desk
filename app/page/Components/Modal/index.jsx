import React from 'react';
import PropTypes from 'prop-types';
import ReactModal from 'react-modal';
import './Modal.scss';

const propTypes = {
  Component: PropTypes.func,
  modalProps: PropTypes.object, // eslint-disable-line
  onModalClose: PropTypes.func,
};

const defaultProps = {
  Component() {},
  modalProps: {},
  onModalClose() {},
};

class Modal extends React.Component {
  render() {
    const {
      Component,
      modalProps,
      onModalClose,
    } = this.props;
    const isOpen = !!Component;

    if (!isOpen) {
      return null;
    }

    return (
      <ReactModal
        className="Modal__container"
        overlayClassName="Modal__overlay"
        contentLabel="Modal"
        isOpen={isOpen}
        ariaHideApp={false}
      >
        <Component {...modalProps} onModalClose={onModalClose} />
      </ReactModal>
    );
  }
}

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;
