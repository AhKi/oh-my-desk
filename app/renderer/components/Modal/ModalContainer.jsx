import { connect } from 'react-redux';
import toJS from 'renderer/components/toJS';
import Modal from 'renderer/components/Modal';
import { langSelector } from 'store/share/status/selectors';
import {
  modalClose,
} from 'actions/modal';
import {
  modalTypeSelector,
  modalPropsSelector,
} from 'store/personal/modal/selectors';

const mapStateToProps = state => ({
  Component: modalTypeSelector(state),
  lang: langSelector(state),
  modalProps: modalPropsSelector(state),
});

const mapDispatchToProps = {
  onModalClose: modalClose,
};

export default connect(mapStateToProps, mapDispatchToProps)(toJS(Modal));
