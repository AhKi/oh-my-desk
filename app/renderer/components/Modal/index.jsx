import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';
import ReactModal from 'react-modal';
import OutsideClickHandler from 'renderer/components/OutsideClickHandler';
import './Modal.scss';

const propTypes = {
  Component: PropTypes.func,
  modalProps: PropTypes.object, // eslint-disable-line
  onModalClose: PropTypes.func,
};

const defaultProps = {
  Component() {},
  modalProps: {},
  onModalClose() {},
};

class Modal extends React.Component {
  render() {
    const {
      Component,
      modalProps,
      onModalClose,
    } = this.props;
    const isOpen = !!Component;
    const {
      modalClass,
      overlayClass,
      activeOutsideClose,
    } = modalProps;

    const modalClassName = cx(modalClass, {
      Modal__container: !modalClass,
    });
    const overlayClassName = cx(overlayClass, {
      Modal__overlay: !overlayClass,
    });

    if (!isOpen) {
      return null;
    }

    return (
      <ReactModal
        className={modalClassName}
        overlayClassName={overlayClassName}
        contentLabel="Modal"
        isOpen={isOpen}
        ariaHideApp={false}
      >
        {activeOutsideClose ? (
          <OutsideClickHandler onOutSideClick={onModalClose}>
            <Component {...modalProps} onModalClose={onModalClose} />
          </OutsideClickHandler>
        ) :
          <Component {...modalProps} onModalClose={onModalClose} />
        }
      </ReactModal>
    );
  }
}

Modal.propTypes = propTypes;
Modal.defaultProps = defaultProps;

export default Modal;
