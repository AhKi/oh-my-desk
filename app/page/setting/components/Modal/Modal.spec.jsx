import React from 'react';
import { shallow } from 'enzyme';

import Modal from './';

describe('<Modal />', () => {
  it('should match to snapshot when render default', () => {
    const wrapper = shallow(<Modal />);

    expect(wrapper).toMatchSnapshot();
  });

  it('should match to snapshot when props.isOpen === false', () => {
    const wrapper = shallow(
      <Modal
        modalType="CONFIRM"
      />,
    );

    expect(wrapper).toMatchSnapshot();
  });
});
