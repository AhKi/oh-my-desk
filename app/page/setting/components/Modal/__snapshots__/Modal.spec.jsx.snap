// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Modal /> should match to snapshot when props.isOpen === false 1`] = `
<Modal
  ariaHideApp={false}
  bodyOpenClassName="ReactModal__Body--open"
  className="Modal__container"
  closeTimeoutMS={0}
  contentLabel="Modal"
  isOpen={true}
  overlayClassName="Modal__overlay"
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
>
  <Connect(Confirm) />
</Modal>
`;

exports[`<Modal /> should match to snapshot when render default 1`] = `""`;
